networking/p2p/src/test/java/tech/pegasys/teku/networking/p2p/connection/ConnectionManagerTest.java

```diff
@@ -204,42 +203,51 @@ public void shouldConnectToKnownPeersWhenStarted() {
     when(network.connect(any(PeerAddress.class))).thenReturn(new SafeFuture<>());
 
     manager.start().join();
+    advanceTimeABit();
 
     verify(network).connect(PEER1);
     verify(network).connect(PEER2);
   }
 
   @Test
+  @Disabled("Shouldn't work, disabled until clarifying")
   public void shouldNotRetryConnectionsToDiscoveredPeersOnFailure() {
     final ConnectionManager manager = createManager();
-    when(discoveryService.streamKnownPeers()).thenReturn(Stream.of(DISCOVERY_PEER1));
+    when(discoveryService.streamKnownPeers())
+        .thenReturn(Stream.of(DISCOVERY_PEER1))
+        .thenReturn(Stream.of(DISCOVERY_PEER1));
     final SafeFuture<Peer> connectionFuture = new SafeFuture<>();
     when(network.connect(any(PeerAddress.class))).thenReturn(connectionFuture);
 
     manager.start().join();
+    advanceTimeABit();
     verify(network).connect(PEER1);
 
     connectionFuture.completeExceptionally(new RuntimeException("Failed"));
 
-    asyncRunner.executeQueuedActions();
+    advanceTimeByInitialInterval();
     verify(network, times(1)).connect(PEER1); // No further attempts to connect
   }
 
   @Test
+  @Disabled("Shouldn't work, disabled until clarifying")
   public void shouldNotRetryConnectionsToDiscoveredPeersOnDisconnect() {
     final ConnectionManager manager = createManager();
-    when(discoveryService.streamKnownPeers()).thenReturn(Stream.of(DISCOVERY_PEER1));
+    when(discoveryService.streamKnownPeers())
+        .thenReturn(Stream.of(DISCOVERY_PEER1))
+        .thenReturn(Stream.of(DISCOVERY_PEER1));
```

Comment:
Possibly this is the issue because the search is now happening more than twice? We'd need to return a new stream from each search so could try making this a `thenAnswer` so it always creates a fresh Stream.

Reply:
It's happening twice, yeah, we'd better create new stream from one statement